satellite/admin-back-office-v1.md

```diff
@@ -0,0 +1,554 @@
+  - Account:
+    - Suspend.
+    - Re-activate.
+    - Delete account.
+- customer support. All the permissions that the _viewer_ role has, and:
+  - Account: All the operations except:
+    - Delete account: When the account isn't clean.
+  - Project: All the operations
+  - Bucket: All the operations.
+
+
+![user roles diagram](admin-back-office-v1/user-roles.png)
+
+The back office configuration allows to specify the relation between roles and groups.
+
+If users belongs to several groups then they will be authorized with the group with the less
+restrictive role.
+
+When the different satellites are hosted in different isolated networks having a single
+authentication proxy requires to establish links between the networks, so the proxy can access to
+all the services. A solution for not having to peer the networks is to have an authentication proxy
+for every satellite, which also mitigates the single point of failure of a single authentication
+proxy.
+
+When using an authentication proxy for each service we have to configure them in a way that users
+don't have to log in more than once when they try to access to a different satellite that the one
+that they logged in for satisfying the _users stories_.
+
+For Storj DCS, we determined that the we want to authenticate the users through Oauth2 with their
+Storj email account, which is hosted by Google Workspace. We are going to authorize assigning them
+to different groups and mapping those groups to the back office roles.
+
+For authenticating the users we will use one [Oauth2-proxy](https://oauth2-proxy.github.io/oauth2-proxy/)
+for each satellite and for the "service" that will show the list of satellites, henceforth "selector
+service".
+
+All the Oauth2-proxies must use the same Oauth2 client credentials for not requiring the users to
+log in when they access to a different satellite or from the "selector service" once they logged in
+successfully in one of them and all the satellites and the "selector service" must be hosted under
+the same base domain or sub-domain. You can see [this proof of concept](https://github.com/ifraixedes/poc-oauth2-proxies-single-auth)
+for how to achieve it.
+
+##### API implementation
+
+We have to review the current satellite administration API to see what endpoints we need to add or
+modify.
+
+We also have to adapt it to allow to perform the operations based on the usr roles specified in the
+_Authentication / Authorization_ section and be able to configure the groups associated to them.
+
+We have to review it to see if we want to add more logging or metrics, considering that wasn't
+created as a main feature as now.
+
+##### Web UI
+
+We have to build a new Web UI from scratch with the Vue framework and the Vuetify component
+framework. The Web UI is already in development and it almost have every component and page that the
+back office requires, you can see it on https://storj.github.io/admin-ui/login.
```

Comment:
This seems to show a lot more information that necessary.

Reply:
I mentioned data aggregation that we don't have for example at the account level.
I'm not sure if we have that information for buckets either.

@stoweandi indicated to @boshevski to remove it (there is a Slack thread discussion where I mentioned), however I don't know if @boshevski had the time to do it.